fix duplicate engine entries

mjkkirschner opened the pull request:
TEMP PR until ironPythondotnet goes into DynamoDS/Dynamo

dnenov committed:
DYN-6493: reset page order when discarding publish package form (#14695)

* reset page order

- goes back to the first page when navigating away from the tab (on cancel only)

* revert namespaces

* revert namespaces

dnenov committed:
DYN-6500: disable cancel button if no changes (#14692)

* disable cancel button of no changes

- if there are no changes, disable the cancel button

* revert namespaces

* revert namespaces

* rename button cancel to reset

dnenov committed:
ui cherrypicks (#14704)

- only UI related cherrypick which should not trigger any test issues

pinzart90 committed:
Webview2 initialization (#14671)

* upate

* Update ViewExtension.cs

* Update DocumentationBrowserView.xaml.cs

* Update NotificationsViewExtension.cs

* update

* update

* Update NotificationCenterController.cs

* Test webview2 (#14670)

* update

* update

* Update DocumentationBrowserView.xaml.cs

* update

* Update DocumentationBrowserView.xaml.cs

* update

* update

---------

Co-authored-by: pinzart <tiberiu.pinzariu@autodesk.com>

* Update NotificationCenterController.cs

* Update DispatcherUtil.cs

* Update CS_SDK.props

* Revert "Update CS_SDK.props"

This reverts commit cfaceb2ee3e5be6500b08ff065642e7248cc69d0.

---------

Co-authored-by: pinzart <tiberiu.pinzariu@autodesk.com>

avidit committed:
DYN-6454 Update smoke test workflow (#14694)

* DYN-6454 Update smoke test workflow

* DYN-6454 Update smoke test workflow

dnenov committed:
enhance dragging experience (#14699)

- created an invisible border to occupy the title bar to help with dragging experience

mjkkirschner committed:
fix duplicate engine entries

avidit committed:
Update workflow event (#14707)

aparajit-pratap commented on src/Libraries/PythonNodeModelsWpf/PythonNode.cs:
```diff
@@ -61,13 +93,7 @@ public void CustomizeView(PythonNode nodeModel, NodeView nodeView)
-                availableEngineNames.Add(nodeModel.EngineName);
-            }
-            availableEngineNames.ForEach(x => AddPythonEngineToMenuItems(x));
+            PythonNodeUtils.GetEngineNames(nodeModel).ForEach(x => AddPythonEngineToMenuItems(x));
```
What if the menu already contains the engine name(s) returned from `GetEngineNames`?

mjkkirschner replied:
I've tried to stop that from happening, but I will add another check in `AddPythonEngineToMenuItems` as well just to be sure.

mjkkirschner replied:
done

dnenov committed:
DYN-6519: clear keywords on reset (#14696)

* clear keywords on reset

- now correctly clears the keyword pills on reset

* namespace revert

* revert namespaces

RobertGlobant20 committed:
DYN-6362 Refresh Packages Tour Bug (#14693)

* DYN-6362 Refresh Packages Tour Bug

I've added three validations for enabling the Dynamo UI when the Packages Tour is closed for preventing the UI Blocked bug.

* DYN-6362 Refresh Packages Tour Bug

Removing empty space

pinzart90 commented on src/DynamoCoreWpf/ViewModels/Menu/PreferencesViewModel.cs:
```diff
@@ -1256,7 +1256,7 @@ internal void HideUnhideNamespace(bool hide, string library, string namespc)
         {
             var options = new ObservableCollection<string> { Res.DefaultPythonEngineNone };
-            foreach (var item in PythonEngineManager.Instance.AvailableEngines)
+            foreach (var item in PythonEngineManager.Instance.AvailableEngines.GroupBy(x=>x.Name).FirstOrDefault().ToList())
```
were there multiple python engines with the same name ?
like 2 versions of Python2 at the same time?

mjkkirschner replied:
yes, thats exactly the bug.

pinzart90 replied:
so if we will support assembly isolation, then we could have the same pacakge but different versions loaded at the same time
So should we enforce uniqueness by appending the version to the engine name (internally at least)?
In the UI we could display the same name but tooltip would clarify which version. Or something like that ..

pinzart90 committed:
Update UnitsCore.csproj (#14709)

Co-authored-by: pinzart <tiberiu.pinzariu@autodesk.com>

dnenov committed:
DYN-6507: crash when uploading package (#14690)

* crash when uploading package

- attempt to fix a crash when uploading a package and clearing the uploadhandle on success

* revert namespaces

* disable dispose during test

* disable CancelCommandClearsAllData test

- disabled CancelCommandClearsAllData test as the command it is supposed to be testing currently is restricted under testing environment as it crashes master-15 (sequentially running of tests) because of thread affinity issue

dnenov committed:
remove duplicate dyf file warning from preview generation process (#14711)

- dynamo would issue a warning and fail to publish a package if an 'unqualified' file is being used, such as a dyf file already under package control
- this check is done when publishing package locally, but I have incorrectly added it to the process of creating a build preview, which stops the process for both local and online submit workflow

pinzart90 committed:
update (#14710)

Co-authored-by: pinzart <tiberiu.pinzariu@autodesk.com>

dnenov committed:
remove package version limitation (#14716)

- now allows package version to start with 0
- cannot have 0.0.0 package version

zeusongit committed:
Localize menu items is Graphic Element Scale dropdown (#14714)

* Fix PostDiff job

* fix

sm6srw committed:
Update build.xml

Temporary set to net6 so we can get a net6 build with latest localization.

sm6srw committed:
Update build.xml

Revert back to net8.

mjkkirschner committed:
remove dynamo sandbox app.config (#14713)

* remove config

* remove autogen stuff

aparajit-pratap committed:
DYN-6527: Fix graph update for primitive input nodes that are first initialized to null  (#14703)

* remove coreclr-ncalc references

* add failing test for dropdown node

* cleanup

* update tests

* attempt initial fix

* cleanup

* update test

* review comments

* add code comments

sm6srw committed:
Bump Dynamo to 3.1 (#14719)

* Bump to 3.1

* Tweak

* Bump year in algorithm to 2023.

* Fix test

* Hardening unittest

mjkkirschner committed:
remove IronPython code and extension (#14706)

* remove legacy completion provider

* remove

* remove obsolete -arguable

tests seem to pass going to merge this and cherry pick to 3.0

mjkkirschner committed:
Merge branch 'master' of https://github.com/DynamoDS/Dynamo into dyn6060

mjkkirschner committed:
review comments
